### PR TITLE
Workaround for a potential bug in XCode.

### PR DESCRIPTION
**Pull request by igilitschenski**
@dymczykm ptal.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,11 +7,27 @@ catkin_simple()
...
+# It is documented at
+#  https://www.mail-archive.com/petsc-users@mcs.anl.gov/msg28472.html
+# and uses gcc as workaround.
+# Consider removal when updating to a new metis version.
+if( (CMAKE_C_COMPILER MATCHES "(.*)Xcode(.*)") AND (CMAKE_C_COMPILER_VERSION MATCHES "7.3(.*)") )
+  message("Use of XCode infeasible due to potential compiler bug. Switching to gcc.")
+  ExternalProject_Add(metis_src
+    URL https://github.com/ethz-asl/thirdparty_library_binaries/raw/master/metis-5.1.0.tar.gz
+    UPDATE_COMMAND ""
+    CMAKE_ARGS -DCMAKE_TOOLCHAIN_FILE=${CMAKE_SOURCE_DIR}/gcc.cmake -DCMAKE_INSTALL_PREFIX:PATH=${CATKIN_DEVEL_PREFIX} -DCMAKE_BUILD_TYPE:STRING=Release -DSHARED:BOOL=TRUE -DGKLIB_PATH:STRING=${CMAKE_CURRENT_BINARY_DIR}/metis_src-prefix/src/metis_src/GKlib
```
Comment (dymczykm) [inline CMakeLists.txt]: would it be possible to pack all the common CMAKE_ARGS into a variable?

**Comment by dymczykm** [comment]
lgtm

File: CMakeLists.txt (hunk 1 above)
Comment (igilitschenski) [inline CMakeLists.txt]: done. lgtm?

**Comment by dymczykm** [comment]
nice, lgtm

**Comment by schneith** [comment]
test this please

**Pull request merged**